aviatrix/resource_aviatrix_site2cloud.go

```diff
@@ -439,16 +439,32 @@ func resourceAviatrixSite2CloudCreate(d *schema.ResourceData, meta interface{})
 		return fmt.Errorf("active_active_ha can't be enabled if HA isn't enabled for site2cloud connection")
 	}
 
+	if s2c.ConnType != "mapped" && s2c.ConnType != "unmapped" {
+		return fmt.Errorf("'connection_type' should be 'mapped' or 'unmapped'")
+	}
+
+	gateway := &goaviatrix.Gateway{
+		GwName: s2c.GwName,
+	}
+
+	gw, err := client.GetGateway(gateway)
+	if err != nil {
+		return fmt.Errorf("couldn't find Aviatrix Gateway %s: %v", s2c.GwName, err)
+	}
+
+	if gw.IsTransit == "yes" {
```

Comment:
If IsTransit has to be added, it should be a boolean instead of a string.

Reply:
changed IsTransit to TransitVpc